### DISCUSSION TITLE
Error in thermal contact based mortar

### DISCUSSION THREAD

**Discussion by jin0123456** [idaholab/moose#28038] [Q&A Modules: Thermal Hydraulics]
Hello,

I met the error in mortar thermal contact,
`The requested mortar interface AutomaticMortarGeneration object does not yet exist!`
And i find it in `MortarData.C` line 123 and 145,
https://github.com/idaholab/moose/blob/61714043f0e9dcbf822da0b852b47ef2c6a08ae2/framework/src/constraints/MortarData.C
```
const AutomaticMortarGeneration &
MortarData::getMortarInterface(const std::pair<BoundaryID, BoundaryID> & boundary_key,
                               const std::pair<SubdomainID, SubdomainID> & /*subdomain_key*/,
                               bool on_displaced) const
{
  if (on_displaced)
  {
    if (_displaced_mortar_interfaces.find(boundary_key) == _displaced_mortar_interfaces.end())
      mooseError(
          "The requested mortar interface AutomaticMortarGeneration object does not yet exist!");

    return _displaced_mortar_interfaces.at(boundary_key);
  }
  else
  {
    if (_mortar_interfaces.find(boundary_key) == _mortar_interfaces.end())
      mooseError(
          "The requested mortar interface AutomaticMortarGeneration object does not yet exist!");

    return _mortar_interfaces.at(boundary_key);
  }
}
```
I want to know about `_displaced_mortar_interfaces.find(boundary_key) == _displaced_mortar_interfaces.end()`,
what's this about? Where is my problem and how can i fix it?

**Comment by GiudGiud** [comment]
Hello

What's missing is that createMortarInterface has not been called beforehand on that interface (two boundaries)
Are you developing or this is straight from a moose input with regular objects?

Guillaume

**Reply by jin0123456** [reply]
I found out the reason. If I set `use_displaced_mesh = true` in `ModularGapConductanceConstraint` object I will get the error or I will not.
So I want to know why would this happen? What is this setting about?
In my case, the deformation of the mesh is relatively large and the distance between two boundaries will decrease during simulation.
How should I use `use_displaced_mesh`? Should I set true or false?

**Reply by GiudGiud** [reply]
@recuero

**Reply by recuero** [reply]
It may be the the user is just missing the creation of the lower dimensional domains? Almost positive MOOSE have examples of these heat transfer mortar  constraints with displaced meshes, maybe through actions.

**Reply by jin0123456** [reply]
I know what i should do now. Much appreciated!
